backend/sqlite/sqlite.go

```diff
@@ -441,40 +444,75 @@ func (be *sqliteBackend) createOrchestrationInstanceInternal(ctx context.Context
 		return errors.New("HistoryEvent must be an ExecutionStartedEvent")
 	}
 
-	// TODO: Support for re-using orchestration instance IDs
-	res, err := tx.ExecContext(
+	var instanceExists bool
+	instanceRes, existErr := tx.QueryContext(
 		ctx,
-		`INSERT OR IGNORE INTO [Instances] (
-			[Name],
-			[Version],
-			[InstanceID],
-			[ExecutionID],
-			[Input],
-			[RuntimeStatus],
-			[CreatedTime]
-		) VALUES (?, ?, ?, ?, ?, ?, ?)`,
-		startEvent.Name,
-		startEvent.Version.GetValue(),
+		`SELECT * FROM [Instances] WHERE [InstanceID] = ?`, 
 		startEvent.OrchestrationInstance.InstanceId,
-		startEvent.OrchestrationInstance.ExecutionId.GetValue(),
-		startEvent.Input.GetValue(),
-		"PENDING",
-		e.Timestamp.AsTime(),
 	)
-	if err != nil {
-		return fmt.Errorf("failed to insert into [Instances] table: %w", err)
+	if existErr != nil {
+		return fmt.Errorf("failed to query from [Instances] table: %w", existErr)
 	}
 
-	rows, err := res.RowsAffected()
-	if err != nil {
-		return fmt.Errorf("failed to count the rows affected: %w", err)
-	}
+	if instanceRes == nil {
+		instanceExists = false
+	} else {
+		instanceExists = true
+	}
+
+	if !instanceExists || startEvent.InstanceExistOption == protos.InstanceExistOption_Throw_IF_EXIST {
```

Comment:
As far as I can tell, this won't actually throw if the instance doesn't exist. The query below is `INSERT OR IGNORE...` which means that the query will just return 0 results if the instance already exists, and we'll return `ErrDuplicateEvent`.

Reply:
Yes, with `ErrDuplicateEvent` send back through grpc, from client side we get an exception as 
<img width="1454" alt="image" src="https://github.com/microsoft/durabletask-go/assets/89094811/e5557667-7cc4-4dae-826c-441b4eeaeb38">
wondering does this count as throw

Reply:
Ah, yes, I think I got confused. My suggestion is that we keep the existing logic as-is except for the `TERMINATE_IF_EXIST` case. Instead of handling `THROW_IF_EXIST` or `SKIP_IF_EXIST` here, we can simply update the caller to check for `ErrDuplicateEvent`. If we get `ErrDuplicateEvent` and the policy is `THROW_IF_EXIST`, then we can return the error as we do today. Otherwise, it the policy is `SKIP_IF_EXIST`, we can just log a warning and return a success. I think it's better to have that logic in the calling code so that we can reduce the amount of special handling the different backends need to implement.